File: classic puzzle-easy/prefix-code.py
#https://www.codingame.com/ide/puzzle/prefix-code
import sys
import math

n = int(input())
dy={}
for i in range(n):
    inputs = input().split()
    b = inputs[0]
    c = int(inputs[1])
    dy[b]=chr(c)
s = input()
l,r=0,0
ans=""
while r<len(s):
    a=s[l:r+1]
    if a in dy:
        ans+=dy[a]
        l=r+1
    r+=1
if l!=len(s):
    print(f'DECODE FAIL AT INDEX {l}')
else: print(ans)

File: classic puzzle-easy/prefix-code.js
//https://www.codingame.com/training/easy/prefix-code
var dy={}
const n = parseInt(readline());
for (let i = 0; i < n; i++) {
    var inputs = readline().split(' ');
    const b = inputs[0];
    const c = parseInt(inputs[1]);
    dy[b]=String.fromCharCode(c);
}
const s = readline();
var ans="";
var r=0,l=0;
while (r<s.length){
    var a=s.slice(l, r+1);
    if (dy.hasOwnProperty(a)){
        ans+=dy[a];
        l=r+1;
    }
    r+=1;
}
if (l!=s.length) console.log("DECODE FAIL AT INDEX "+l);
else console.log(ans)
